### ISSUE TITLE
No need to replace the nyan cat?

### ISSUE THREAD

**Issue by paramat** [minetest/minetest_game#1685]
It seems that very many people dislike the addition of the PB&J Pup, not necessarily for it's design and implementation, which were good, but because a replacement that is another character with a differerent type of trail feels like the wrong thing to do.
I feel this way too. Nothing could replace the nyancat and any replacement added, however good, is almost inevitably going to not feel right.

I expect those who loved the nyancat will install the nyancat mod to continue using it and will disable PB&J Pup generation, i will be too. Those who are not keen on the nyancat will not need a replacement.

So i feel bad for suggesting this, but, my intuition is telling me the addition of a similar replacement was a mistake, we should not try to replace the nyancat at all. I feel we should remove the PB&J mod from MTGame. Luckily they are rare so there will only be a few unknown nodes around to dig, or perhaps they can be aliased to air.

I want to be clear that i feel the design was good and the implementation by sofar was excellent. However i also want to apologise to VanessaE and sofar for suggesting this.

**Comment by ExeterDad** [comment]
I'm not a fan of the pup only because I am a fan of the cat. To me the cat is something special to Minetest much like MESE is special.
However... on our server, we have decided to embrace the pup and have it installed alongside the Nyan cat. We wanted to keep the cat for obvious reasons as we didn't want builds breaking and players were concerned about Minetests need to remove the cat to protect itself.

Since the announcement that the PB&J pup is now live on the server, the reaction overall has been positive and I see mention of it several times a day. Most of the mentions are because no one has been able to find them so far. People are searching for the pups. I'm sure it's because they are anxious to see the thing in all it's barking glory. But also they are actively searching because we put up a bounty of 500 gold ingots for the first pup found.
maikerumine (a well known player/admin/Youtuber) has even visited our server while recording a Youtube video with PB&J pup in the title. He was on a quest to find the first pup in the wild. Naturally he wasn't able to find the pup during the length of the video.

Of course I do hear lot's of people saying they hate the pup, and don't want to see the Nyan removed. But if you pay attention to who saying these things, you will note it's the seasoned Minetesters who are naturally set in their ways and are likely reluctant to embracing change. We see that behavior every day :)
I guess the point of this ramble is to reflect that the majority of our players are responding to the pup in a positive way. Even if I feel differently and am planning to keep the Nyan cat alive and well until somebody important forces me not to.

**Comment by Wuzzy2** [comment]
Actually, I don't have so strong feelings about Nyan Cat and PB&J Pup anymore.
These blocks are very rare and only meant as “joke blocks” anyway. So these blocks have very little relevance.
I don't care if those are removed, but I am also not opposed if they are kept.
Just try to make sure you don't break existing worlds without reason.

I feel the discussion about this whole thing has goon far too long for now, and I start to get tired of it. xD

**Comment by C1ffisme** [comment]
@Wuzzy2 Agreed. There isn't any real point to the blocks, besides their history in Minetest. Considering we don't even have a 1.0.0 yet, I don't think we should get too nostalgic about certain points in this game's development history.

**Comment by sofar** [comment]
> I don't care if those are removed

yup, I feel the same way.

I did feel inclined to make a PR originally to resolve the stalemate (nobody else submitted code to resolve a potential legal problem) but at this point there are no legal problems that I can find, and so, you couldn't poke me to make another PR to change it.

But if someone does change it, sure, have at it. :)

**Comment by XtremeHacker** [comment]
I think that PB J Pup is OK on It's own, but feels like a copy + paste type replacement for Nyan cat, a completely unique replacement would be nice, any ideas on some totally new multi-block spelunker?

**Comment by dsohler** [comment]
> I expect those who loved the nyancat will install the nyancat mod to continue using it and will disable PB&J Pup generation, i will be too. Those who are not keen on the nyancat will not need a replacement.

I 100 percent feel the same.

> So i feel bad for suggesting this, but, my intuition is telling me the addition of a similar replacement was a mistake, we should not try to replace the nyancat at all.

So while the replacement option itself is a good thing I think it adds unnecessary complexity to the *PB&J Pub* mod which could be avoided when not having an option to replace/alias the *Nyan Cat* mod's nodes by configuration option but provide it as external mod that by coding replaces/aliases those nodes. So the *PB&J Pub* mod doesn't need any configuration options at all.

> I feel we should remove the PB&J mod from MTGame.

Yes. But provide it via the forum as free (as in free beer AND free speech) alternative to the *Nyan Cat* mod.

**Comment by tenplus1** [comment]
Free replacement (Nyan Cat AND PB&J Pup) - https://github.com/tenplus1/pbj_pup

**Comment by sofar** [comment]
> So while the replacement option itself is a good thing I think it adds unnecessary complexity to the PB&J Pub mod which could be avoided when not having an option to replace/alias the Nyan Cat mod's nodes by configuration option but provide it as external mod that by coding replaces/aliases those nodes. So the PB&J Pub mod doesn't need any configuration options at all.

It doesn't. Those extra options are entirely optional and you don't ever need to touch them. Just install a different mod that provides the necessary aliases and you're on your way.

**Comment by dsohler** [comment]
> Those extra options are entirely optional and you don't ever need to touch them.

Isn't being optional the point of options? :smile:  any yes, you don't have to use them. But the mod has to check them.

**Comment by sofar** [comment]
>> Those extra options are entirely optional and you don't ever need to touch them.

> And yes, you don't have to use them. But the mod has to check them.

[deleted profanity laden reply]

[deleted more polite reply questioning reasoning]

[deleted profanity filled reply, again]

sigh

Look, @dsohler. You are telling me that you'd rather see JoeShmoe perform the following actions:

1) click a zip link found in a forum post for a mod that installs 2 lines of code
2) figure out how to unzip it because windows (android, etc) don't ship a zip client
3) unzip it
4) copy it to ~/My Games/
5) wonder why it doesn't work
6) copy it to ~/Program Files/Minetest
37) finally find the right location where minetest is installed
38) wonder why it still doesn't work
39) enable the mod in the mod config setting
40) wake up from this nightmare

instead of

1) go to settings
2) search for `nyan`
3) enable the options listed.

If you're convinced that you're so right about this, then go ahead, submit a pull request to remove the options.

**Comment by HybridDog** [comment]
step 9 doesn't work on windows

**Comment by dsohler** [comment]
> [deleted profanity laden reply]
>
> [deleted more polite reply questioning reasoning]
>
> [deleted profanity filled reply, again]

Oh, you're so funny and sarcastic. I bet everyone now is proud of your maturity.

Try to think for a second what kind of complexity I was talking about.

**Comment by sofar** [comment]
> Try to think for a second what kind of complexity I was talking about.

Make a patch already.

**Comment by dsohler** [comment]
I'd patch it completely out. Nothing will be left neither of *Nyan Cat* nor of *PB&J Pub*. No nodes, no aliases, no configuration options - nothing.

If this is what you want, I'd be happy to make a pull request.

**Comment by paramat** [comment]
The complexity of PB&J settings is no issue, it's more that this kind of mod should be a separate mod from MTGame due to being so specialised. I know that nyancat was part of MTGame but that was to support already placed nodes, there is no need to add another new mod of this type, after all:

> I expect those who loved the nyancat will install the nyancat mod to continue using it and will disable PB&J Pup generation, i will be too. Those who are not keen on the nyancat will not need a replacement.

Removing nyancat and adding PB&J was done with a single commit, i think it should have been 2 separate commits.

Those who like the nyancat (a huge number of players, maybe a majority) will now simply be installing it as a separate mod instead of a built-in mod, they do not need a second mod of this type added.
Those who do not care for the nyancat will not be wanting a similar mod added.

So ideally i feel we should have done what i originally suggested: remove the nyancat mod, do not add any aliases, in order to preserve the nodes so that when the mod is added back in the unlnown nodes return to nyancats.
Then those who don't re-install the mod will have some unknown nodes to clear up (not many luckily), but this is preferable to them than a nyancat parody taking it's place. Otherwise they can add a simple mod that aliases their existing nyancat nodes to valuable ores or stone or whatever they want.

Those who like the PB&J mod can add it as a separate mod, it can be stored in Minetest-Mods.

**Comment by bigfoot547** [comment]
@dsohler You are being immature.

**Comment by paramat** [comment]
> maikerumine (a well known player/admin/Youtuber) has even visited our server while recording a Youtube video with PB&J pup in the title. He was on a quest to find the first pup in the wild.

Yes i watched that video and he describes how he feels about it, he doesn't like it either.

As a separate mod taken isolation it is fine, likable  and well made, but to be honest in context as a replacement for the nyancat i strongly dislike it and consider it a design decision disaster. it's also more code to maintain in MTG while being unnecessary.

**Comment by paramat** [comment]
PR #1695

**Comment by dsohler** [comment]
That’s by far the cleanest and best solution :+1:

**Comment by paramat** [comment]
#1695 merged.

**Comment by stas730** [comment]
Reference: https://github.com/minetest/minetest_game/issues/2099
Nyan Cat was removed some time ago. Many players have installed nyancat mod because they didn’t want Nyan Cat to be removed.
Nyan Cat can be added again, but name must be changed to avoid trademark issues. "Nyā Cat" is example ("nyā" is Japanese word, so it can’t be trademarked).

**Comment by VanessaE** [comment]
Technically the name "Nyan Cat" should have been just fine, because THAT is the name of the audio/video meme created by someone else *after* the Pop Tart Cat animated GIF was published, and named in part for the song used in the video... but somehow the original Cat artist was able to trademark the name.  On the other hand, the phrase "Pop Tart Cat" is NOT trademarked (I checked).  No surprise there, since "Pop Tart[s]" is a registered trademark of Kellogg's.  Oddly, the original Nyan Cat imagery does not appear to be trademarked, but copyright covers that anyway.

While one cannot trademark just any generic word and expect the it to stand for very long, one *can* trademark a unique word or design, or a generic phrase if combined with something else unique (such as the aforementioned "Kellogg's", especially when rendered in its usual red, cursive font, or Nike's old "Just do it" slogan, when used in the context of shoes).

I'd prefer that the Cat be added back in, but calling it "Nyā Cat" would be too similar to "Nyan Cat".  You'd have to choose a name that has no potential for confusion, especially in the gaming market.  I drew the last set of Cat images from scratch, trying to keep it familiar to MT players, but with enough differences that no one could reasonably confuse my version for the original.

**Comment by stas730** [comment]
What if replace the names with "Cat" and "Rainbow" (or "Cat Rainbow")?

**Comment by stas730** [comment]
^ Also, remove "PB&J" from the names and add PB&J Pup again.

**Comment by VanessaE** [comment]
Well simply calling it Rainbow Cat *might* be okay, but a google search for the phrase (in quotes) gives many references to Nyan Cat.  However, that phrase is *not* trademarked (there is one entry, but it is abandoned).

**Comment by VanessaE** [comment]
Why rename PB&J Pup?  I made sure that phrase was not trademarked, and posted a clear and unambiguous claim to it.

**Comment by stas730** [comment]
@VanessaE, from your message: "**A google search** shows that that term (with or without the "&") appears to be unclaimed (and the few results I did find were just the usual search engine gibberish, like two hash tags close together, or license plate suggestions). Therefore, if someone thinks this is a good idea, I'll try to draw appropriate images. To that end, I lay claim to the terms "PB&J Pup" and "pbj_pup" and the images I described above, as of today's date.".
It looks like Google LLC is the name owner.
The rule: **Never trust** network‐connected computers or SaaS to check trademarks.

**Comment by benrob0329** [comment]
``Strudal Cat`` - @VanessaE
``Cyan Cat`` - @paramat
``Baguette Cat`` - Me

<details>
<summary>Joke</summary>
``Chicken Sandwich Cat`` - Also Me
</details>

**Comment by VanessaE** [comment]
@stas730, we're going way off topic here, but the google search at the time was just to get a general idea of uniqueness, just so I didn't end up using a term that's already used for something else unique and popular.  I found nothing relevant at the time, and I did a trademark search directly via USPTO's website, using multiple variations of the term.  The phrase PB&J Pup, with or without "&", had no trademark.

Just to be sure, I checked again just now.  USPTO records no relevant trademark, and a google search shows that while there are some people using the term currently, none can reasonably be considered commercial core-business (the closest ones are some bakery run by a 14 year old, a rescue-animal-related chocolate bar which has the phrase as part of its name, and a really cute piece of furry-related artwork).  I'm certain that none of those predates my work -- if they did, I'm sure I would have noticed back then -- and none appear to remotely be in the same market anyway.  The top results all point to Minetest in some way.

As far as I'm concerned, that constitutes as much due diligence on my part as I am capable of, so I reiterate my original claim.  The terms "PBJ Pup", "PB&J Pup", and "pbj_pup" (and the like) and the associated imagery are my property in the gaming market, and all related markets, period.  But just as I said before, they are 100% free to use for any Minetest-related purpose (though if someone makes money off of it, as unlikely as it is, I'd like a cut).

Now all that aside, @benrob0329, I'm not sure Struedel/Strudel Cat would work, since that's the name of an existing spoof.  Any other flat pastry would probably work just as well. (Blintz Cat? Crêpe Cat? :stuck_out_tongue: )
